diff -ruN alsa-lib-1.0.18.orig/src/pcm/pcm_direct.c alsa-lib-1.0.18/src/pcm/pcm_direct.c
--- alsa-lib-1.0.18.orig/src/pcm/pcm_direct.c	2008-10-29 06:42:10.000000000 -0600
+++ alsa-lib-1.0.18/src/pcm/pcm_direct.c	2009-01-07 15:03:25.000000000 -0700
@@ -425,12 +425,20 @@
 		return ret;
 	}
 	
+#if defined(HAVE_FORK)
 	ret = fork();
+#else
+	ret = vfork();
+#endif
 	if (ret < 0) {
 		close(dmix->server_fd);
 		return ret;
 	} else if (ret == 0) {
+#if defined(HAVE_FORK)
 		ret = fork();
+#else
+		ret = vfork();
+#endif
 		if (ret == 0)
 			server_job(dmix);
 		_exit(EXIT_SUCCESS);
